### PR TITLE
make jvm heap usage a dynamic setting

### PR DESCRIPTION
**Pull request by jackiehanyang**
### Description
make jvm heap usage a dynamic setting

### Issues Resolved
https://github.com/opensearch-project/anomaly-detection/issues/1213

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/timeseries/breaker/MemoryCircuitBreakerNumericSetting.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import static java.util.Collections.unmodifiableMap;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.timeseries.settings.DynamicNumericSetting;
+
+public class MemoryCircuitBreakerNumericSetting extends DynamicNumericSetting {
```
Comment (kaituo) [inline src/main/java/org/opensearch/timeseries/breaker/MemoryCircuitBreakerNumericSetting.java:16]: Can you move your code to ADNumericSetting ? We don't need to create a new class for each setting. Also, we covered setting init in TimeSeriesAnalyticsPlugin. Once you moved your code to ADNumericSetting, we don't need another call to initialize memory circuit breaker setting.

**Review by kaituo — COMMENTED** [review]
Will wait for you to do an e2e testing.

File: src/main/java/org/opensearch/timeseries/breaker/MemoryCircuitBreakerNumericSetting.java (hunk 1 above)
Comment (jackiehanyang) [reply src/main/java/org/opensearch/timeseries/breaker/MemoryCircuitBreakerNumericSetting.java]: moved the logic to `ADNumericSetting` file in the latest revision

**Comment by jackiehanyang** [comment]
> Will wait for you to do an e2e testing.

Will do an e2d test when both 2.15 changes are done. Tracking it here - https://github.com/opensearch-project/anomaly-detection/issues/1213

**Review by kaituo — APPROVED** [review]

**Comment by jackiehanyang** [comment]
did e2e testing and verified result with Kaituo offline

**Pull request merged**